common/src/main/java/com/wynntils/wc/utils/lootrun/LootrunUtils.java

```diff
@@ -284,6 +266,23 @@ private static void renderPoints(
         }
     }
 
+    private static BlockValidness checkBlockValidness(Level level, ColoredPoint point) {
+        BlockValidness state = BlockValidness.INVALID;
+        Iterable<BlockPos> blocks = getBlocksForPoint(point);
+
+        for (BlockPos blockInArea : blocks) {
+            BlockState blockStateInArea = level.getBlockState(blockInArea);
+            if (blockStateInArea.is(Blocks.BARRIER)) {
+                state = BlockValidness.HAS_BARRIER;
```

Comment:
should be an auto return,

Reply:
No, there could be a vaild block when there is a barrier.

Reply:
doesn't this create conflicts based off if the valid block or the barrier occurs first?

Reply:
I followed what the original code did. I think if we have a valid block, we don't need to care about barriers, but we still search for barriers in case there is no valid block, as the rendering process is different for all outcomes

Reply:
Original code: 
```java
boolean barrier = false;
boolean validBlock = false;
for (BlockPos blockInArea : blocks) {
    BlockState blockStateInArea = level.getBlockState(blockInArea);
    if (blockStateInArea.is(Blocks.BARRIER)) {
        // System.out.println("barrier");
        barrier = true;
    } else if (blockStateInArea.getCollisionShape(level, blockInArea) != null) {
        validBlock = true;
    }
}

 if (validBlock) {
      disable = false;
      if (disabled) {
          consumer = source.getBuffer(RenderType.lineStrip());
          disabled = false;
      }
      for (Point location : toRender) {
          Vec3 rawLocation = location.vec3();
          int pathColor = location.color();
          consumer.vertex(last, (float) rawLocation.x, (float) rawLocation.y, (float)
                          rawLocation.z)
                  .color(pathColor)
                  .normal(0, 0, 1)
                  .endVertex();
      }
      toRender.clear();
  } else if (barrier) {
      disable = true;
      pauseDraw = true;
      toRender.clear();
  } else if (disable) {
      pauseDraw = true;
  } else {
      toRender.add(loc);
      continue;
    }
}
                                    ```

Reply:
ah